src/collectors/collector_worker.go

```diff
@@ -93,13 +93,18 @@ func (cw *CollectorWorker) Run(device configuration.DeviceConfig) (string, error
 
 	// handle client timeouts
 	clientConnected := make(chan struct{})
-	go func() {
-		connection, err = utils.SSHClient(*s)
+
+	go func(err *error) {
+		connection, *err = utils.SSHClient(*s)
 		close(clientConnected)
-	}()
+	}(&err)
+
 	// wait for client
 	select {
 	case <-clientConnected:
+		if err != nil {
+			return "", cw.collectFailure(device, err)
```

Comment:
I have not seen the output of the failure here. Would it make sense to add some additional text for context with the err here? Or `log.Printf("SSHClient Connect Error: %s",  err.Error())`

Obviously don't need to double up error messages, but if there isn't already some context identifier somewhere in the failure path, it could be useful for future debugging.

Reply:
The SSHClient errors make this clear I think.